stdlib/src/builtin/builtin_slice.mojo

```diff
@@ -75,35 +65,33 @@ struct Slice(Sized, Stringable, EqualityComparable):
         self.step = 1
 
     @always_inline("nodebug")
-    fn __init__[
-        T0: AnyRegType, T1: AnyRegType, T2: AnyRegType
-    ](inout self, start: T0, end: T1, step: T2):
+    fn __init__(
+        inout self,
+        start: OptionalReg[Int],
+        end: OptionalReg[Int],
+        step: OptionalReg[Int],
+    ):
         """Construct slice given the start, end and step values.
 
-        Parameters:
-            T0: Type of the start value.
-            T1: Type of the end value.
-            T2: Type of the step value.
-
         Args:
             start: The start value.
             end: The end value.
             step: The step value.
         """
-        self.start = _default_or(start, 0)
-        self.end = _default_or(end, _int_max_value())
-        self.step = _default_or(step, 1)
+        self.start = start
+        self.end = end
+        self.step = step.value() if step else 1
 
     fn __str__(self) -> String:
         """Gets the string representation of the span.
 
         Returns:
             The string representation of the span.
         """
-        var res = str(self.start)
+        var res = str(self.start.value()) if self.start else ""
```

Comment:
Are there any tests for `slice.__str__`? If not, could you please add some?

Reply:
Yes there are https://github.com/modularml/mojo/blob/main/stdlib/test/builtin/test_slice.mojo#L78-L82